### ISSUE TITLE
CC064 invalid error

### ISSUE THREAD

**Issue by SJMakin** [code-cracker/code-cracker#418]
I believe that an error is being generated incorrectly

The code:

    Public Shared Function ToEnum(Of TEnum)(ByVal value As Object) As TEnum
        Dim rawValue As String = value.ToString()
        Dim result As TEnum
        Try
            result = DirectCast([Enum].Parse(GetType(TEnum), rawValue), TEnum)
        Catch ex As ArgumentException
            result = Nothing
        End Try

        Return result
    End Function

The issue is that 'GetType(TEnum)' is being shown as CC0064 Object reference not set to instance of an object, when it is valid.

@adraut is working on it.

**Comment by giggio** [comment]
Bug confirmed. `CC0064` is incorrectly being raised on `Enum.Parse`. A simpler repro:

````vb
    Public Sub Foo()
        [Enum].Parse(GetType(String), "")
    End Sub
````

**Comment by adraut** [comment]
@giggio Should I add to the analyzer to also cover the TryParse method or should that be done in a different issue?

**Comment by giggio** [comment]
@adraut Seems like a different issue to me.

**Comment by mc0re** [comment]
Same goes for other Parse() functions (Integer.Parse, Double.Parse, Byte.Parse).

**Comment by giggio** [comment]
@mc0re This should not affect other parse functions after the fix on #418. We have not yet published this update. I am thinking of releasing an RC2 before 1.0, so this bugs already fixed don't have to be waiting.
